### ISSUE TITLE
E999 Syntax error on match statement with python 3.10

### ISSUE THREAD

**Issue by leddy231** [charliermarsh/ruff#1783]
Incredible tool so far, working on replacing all of pylint in our monorepo with ruff right now, and using the vscode extension.

*Bug:*
Ruff seams to not be able to handle python 3.10 match cases? 
Got it down to a very small working example, even specifying py310 in pyproject.toml or from command line does not work.
This example is just one python file, no pyproject.toml or ruff.toml.

![image](https://user-images.githubusercontent.com/2323691/211818390-0eea2e8b-a365-4aa5-a4b6-284bb583d247.png)

Here is the small code snippet.
```python
import os # should trigger "unused import"

match "x":
  case "x":
    print("x")
```

Edit:
Here is a Dockerfile that re-creates the bug, so it is nothing weird on my machine
```Dockerfile
FROM python:3.10-alpine

RUN pip install ruff
RUN echo "match 'x':"     >> test.py
RUN echo "  case 'x':"    >> test.py
RUN echo "    print('x')" >> test.py
RUN python test.py
RUN ruff --target-version "py310" test.py
```

**Comment by messense** [comment]
Duplicate of https://github.com/charliermarsh/ruff/issues/282

**Comment by leddy231** [comment]
And i though i was good at searching python issues... :sweat_smile:

**Comment by charliermarsh** [comment]
All good :) We're working on it!

**Comment by charliermarsh** [comment]
Fixed as of v0.0.250 which included `match` statement support.

**Comment by SirDank** [comment]
@charliermarsh got this on `Python 3.12.8`

**Comment by MichaReiser** [comment]
@SirDank see https://github.com/astral-sh/ruff/issues/16417
